miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+    # "uid": {"siid": 8, "piid": 8}, # str
+    "auto_boost": {"siid": 8, "piid": 9},  # bool (auto boost on carpet)
+    "forbid_mode": {"siid": 8, "piid": 10},  # str
+    "water_level": {"siid": 8, "piid": 11},
+    # "siid8_13": {"siid": 8, "piid": 13}, # no-name: (uint32, unit: seconds) (acc: ['read', 'notify'])
+    # "siid8_14": {"siid": 8, "piid": 14}, # no-name: (uint32, unit: none) (acc: ['read', 'notify'])
+    "clean_counts": {"siid": 8, "piid": 18},
+    # "siid8_19": {"siid": 8, "piid": 19}, # no-name: (uint32, unit: seconds) (acc: ['read', 'notify'])
+    "double_clean": {"siid": 8, "piid": 20},
+    "edge_sweep": {"siid": 8, "piid": 21},
+    "led_switch": {"siid": 8, "piid": 22},
+    "lidar_collision": {"siid": 8, "piid": 23},
+    "station_key": {"siid": 8, "piid": 24},
+    "station_led": {"siid": 8, "piid": 25},
+    "current_audio": {"siid": 8, "piid": 26},  # str (example: girl_en)
+    "progress": {"siid": 8, "piid": 28},
+    # "station_type": {"siid": 8, "piid": 29}, # uint32
+    # "voice_conf": {"siid": 8, "piid": 30},
+    # "switch_status": {"siid": 2, "piid": 10},
+    "volume": {"siid": 9, "piid": 1},
+    "mute": {"siid": 9, "piid": 2},
+}
+
+
+class ChargingState(Enum):
+    Unknown = -1
+    Charging = 1
+    Discharging = 2
+    NotChargeable = 4
+
+
+class CleaningMode(Enum):
+    Unknown = -1
+    Silent = 1
+    Basic = 2
+    Strong = 3
+    FullSpeed = 4
+    Sweep = 0
+
+
+class SweepType(Enum):
+    Unknown = -1
+    Sweep = 0
+    Mop = 1
+    MopAndSweep = 2
+
+
+class SwitchStatus(Enum):
+    Unknown = -1
+    Open = 1
+
+
+class PathMode(Enum):
+    Unknown = -1
+    Normal = 0
+    YMopping = 1
+    RepeatMopping = 2
+
+
+class WaterLevel(Enum):
+    Unknown = -1
+    First = 1
+    Second = 2
+    Three = 3
+    Fourth = 4
+    Mop = 0
+
+
+class SweepMode(Enum):
+    Unknown = -1
+    Total = 1
+    Area = 2
+    Curpoint = 3
+    Point = 4
+    Smart = 7
+    AmartArea = 8
+    DepthTotal = 9
+    AlongWall = 10
+    Idle = 0
+
+
+class FaultStatus(Enum):
+    Unknown = -1
+    NoFaults = 0
+    LowBatteryFindCharger = 1
+    LowBatteryAndPoweroff = 2
+    WheelRap = 3
+    CollisionError = 4
+    TileDoTask = 5
+    LidarPointError = 6
+    FrontWallError = 7
+    PsdDirty = 8
+    MiddleBrushFatal = 9
+    SidBrush = 10
+    FanSpeedError = 11
+    LidarCover = 12
+    GarbageBoxFull = 13
+    GarbageBoxOut = 14
+    GarbageBoxFullOut = 15
+    PhysicalTrapped = 16
+    PickUpDoTask = 17
+    NoWaterBoxDoTask = 18
+    WaterBoxEmpty = 19
+    CleanCannotArrive = 20
+    StartFormForbid = 21
+    Drop = 22
+    KitWaterPump = 23
+    FindChargerFailed = 24
+    LowPowerClean = 25
+
+
+class DeviceStatus(Enum):
+    Unknown = -1
+    Dormant = 1
+    Idle = 2
+    Paused = 3
+    Sweeping = 4
+    GoCharging = 5
+    Charging = 6
+    Error = 7
+    Rfctrl = 8
+    Fullcharge = 9
+    Shutdown = 10
+    FindChargerPause = 11
+
+
+class RoidmiVacuumStatus(DeviceStatusContainer):
```

Comment:
```suggestion
class RoidmiVacuumStatus(DeviceStatus):
```
please extend this directly from the DeviceStatus for now. Also, please use the names and return types consistent to vacuumcontainers.py (e.g., `ConsumableStatus` for consumables), as we want to move slowly towards common interfaces among device types for interoperability.